fix: have icons use the new render management system

BeksOmega opened the pull request:
<!--
  - Thanks for submitting code to Blockly!  Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## The basics

<!-- TODO: Verify the following, checking each box with an 'x' between the brackets: [x] -->

- [X] I branched from develop
- [X] My pull request is against develop
- [X] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [X] I ran `npm run format` and `npm run lint`

## The details
### Resolves

<!-- TODO: What Github issue does this resolve? Please include a link. -->
Work on #7068 

### Proposed Changes

<!-- TODO: Describe what this Pull Request does.  Include screenshots if applicable. -->
1) Adds a new method to the render management system to immediately trigger any queued renders. This should only be used in cases where it's necessary for backwards compatibility (like this one).
2) Switches icon methods over to use the render management system (with the immediate trigger).
3) Adds dummy promises to the return value of all `setBubbleVisible` methods.

### Reason for Changes

<!--TODO: Explain why these changes should be made.  Include screenshots if applicable. -->
1) We need adding icons to trigger a render immediately so that if the bubbles are set to be visible immediately, they are positioned correctly. Eventually we want to move this to be promises-based (so setting the visibility will wait for any queued renders to finish first). But that is a breaking change.
2) This gets us one step closer to tearing out the old render management system.
3) This makes it easier for external devs to transition to the new promises based system.

### Test Coverage

<!-- TODO: Please create unit tests, and explain here how they cover
           your changes, or tell us how you tested it manually. If
           your changes include browser-specific behaviour, include
           information about the browser and device that you used for
           testing. -->

All tests pass! Also quickly tested `block.setCommentText('test'); block.getIcon('comment').setBubbleVisible(true);` in the browser console to reassure myself that it was indeed working.

rachel-fenichel commented on core/block_svg.ts:
```diff
@@ -1012,8 +1012,8 @@ export class BlockSvg
-      // TODO: Change this based on #7068.
-      this.render();
+      this.queueRender();
+      renderManagement.triggerQueuedRenders();
```
You can move the trigger into `queueRender` since you're always doing them together.

rachel-fenichel replied:
I take it back--in a later PR you aren't doing them together.

BeksOmega replied:
In this PR yes! But there are other places throughout the `BlockSvg` class where I don't trigger an immediate render. I.e. I use the render management system properly :P

cpcallen reviewed (approved):
Make sure that this PR still has a `deprecation` tag once it's been submitted: one of the bots seems to like to try to remove that tag.

Not for this PR, but: given similarity of code have you considered introducing a `BubbleIcon` intermediate class between `Icon` and `Comment`/`Mutator`/`WarningIcon`?

BeksOmega commented:
Maribeth actually requested on the next pull request that I remove the dummy promises, and we just cleanly break people later. So I've reverted that change and removed the deprecation notice!
